lewis/adapters/stream.py

```diff
@@ -128,18 +167,66 @@ def __init__(self, target_method, regex, regex_flags=0, argument_mappings=None,
         self.return_mapping = return_mapping
         self.doc = doc
 
-    def map_arguments(self, arguments):
-        """
-        Returns the mapped function arguments. If no mapping functions are defined, the arguments
-        are returned as they were supplied.
+    @property
+    def patterns(self):
+        return [self.pattern.pattern]
 
-        :param arguments: List of arguments for bound function as strings.
-        :return: Mapped arguments.
-        """
-        if not self.argument_mappings:
-            return arguments
+    def process_request(self, request, target):
+        match = self.pattern.match(request)
 
-        return [f(a) for f, a in zip(self.argument_mappings, arguments)]
+        if not match:
+            raise CanNotProcessException()
+
+        args = self.map_arguments(match.groups())
+        func = getattr(target, self.member)
+
+        return self.map_return_value(func(*args))
+
+
+class Var(BaseCommand):
+    def __init__(self, target_member, read_pattern=None, write_pattern=None,
+                 argument_mappings=None, return_mapping=lambda x: None if x is None else str(x),
+                 doc=None):
+        self.member = target_member
+
+        self._patterns = {key: re.compile(pattern) for key, pattern in
+                          zip(('read', 'write'), (read_pattern, write_pattern)) if
+                          pattern is not None}
+
+        if 'read' in self._patterns and self._patterns['read'].groups != 0:
```

Comment:
The first part of the check wasn't there, so it would have failed for cases with only a read or write pattern. Collecting these in a dict solves the problem of `patterns` as well, no filter required there anymore, because read/write pattern are not in the dict.